lib/client/jsx/components/inputs/select_input.jsx

```diff
@@ -6,17 +6,26 @@ const NoneOption = (showNone) => (
   : null
 );
 
-const Option = (v) => (
+const Option = (v,i) => (
```

Comment:
I find this function very hard to understand. I think if the values where transformed into the same types before being passed to Options it would be more clear.

Reply:
It was intended to be an interface allowing the user to optionally specify labels for their values. I.e., most of the time I am happy with <SelectInput values={ [ 'a', 'b', 'c' ] }/>, but sometimes I want labels, e.g. <SelectInput values={ [ { value: 'a', label: 'Option A' }, ... ] }/> - so a value when it comes from props.values will either be an object or not.

Since it comes as a prop, seems like I'd have to transform the values every time I touch them (in render(), onChange(), etc.) to make this consistent, which seems wrong. Looks like 'key' is unnecessary, so I can ditch that.